packages/components/src/custom-select-control-v2/legacy-component/index.tsx

```diff
@@ -76,7 +77,12 @@ function CustomSelectControl( props: LegacyCustomSelectProps ) {
 					value={ name }
 					children={ __experimentalHint ? withHint : name }
 					style={ style }
-					className={ className }
+					className={ clsx(
+						// Legacy classnames
+						'components-custom-select-control__item',
+						!! __experimentalHint && 'has-hint',
```

Comment:
Consumers of the V1 component expect these classnames and use them to style specifically the `hint` part (you can quickly see by searching for `.has-hint` in the codebase).

I'll see if I can bring in more changes to the hint so that these overrides are not necessary in the editor, but it's probably sensible to leave those classnames there, at least for the legacy adapter.